lib/lltz_ir/dsl.ml

```diff
@@ -86,7 +86,7 @@ module T = struct
   let chain_id_prim = LE.{ desc = Prim (LP.Chain_id, []); range = dummy  }
   let level = LE.{ desc = Prim (LP.Level, []) ; range = dummy }
   let now = LE.{ desc = Prim (LP.Now, []); range = dummy  }
-  let self ?opt () = LE.{ desc = Prim (LP.Self opt, []); range = dummy  }
+  let self opt () = LE.{ desc = Prim (LP.Self opt, []); range = dummy  }
```

Comment:
Note need for the unit parameter, unless a higher-function is expecting this kind of type?

Reply:
resolves after rebase